models/book.go

```diff
@@ -0,0 +1,39 @@
+package models
+
+import (
+	"fmt"
+	"strings"
+	"time"
+)
+
+type Book struct {
+	Id             string     `json:"index._id" binding:"required"`
+	Title          string     `json:"title" binding:"required"`
+	AuthorName     string     `json:"author_name"  binding:"required"`
+	Price          float64    `json:"price"  binding:"required"`
+	EbookAvailable bool       `json:"ebook_available"  binding:"required"`
+	PublishDate    CustomTime `json:"publish_date" binding:"required"`
+}
+
+type CustomTime time.Time
+
+const ctLayout = "2006-01-02"
```

Comment:
You use this "2006-01-02" in another place as well.
Make it public and use the const there too.
ctLayout is a bad name, what about `timeFormat`?